### DISCUSSION TITLE
Ifan03 schematic help

### DISCUSSION THREAD

**Discussion by djpimp2010** [arendst/Tasmota#12790] [Q&A] [answered]
Hi guys, absolutely adore this firmware and this group is awesome!! Love the support you guys give people on here so thanks!

Now, onto the task.

I have multiple ifan03’s, and one of them I need to convert.
I know I can win a GPIO pin or two by disabling RF support. I dont use RF for ifans so thats all good.

I have a 3.3/5v powered microcrontroller, that converts a PWM signal into a 240v dimmer.

Ideally I can steal a 5v supply from the ifan board, so that the pwm controller can stay hidden in the ifan case (my install space is severely limited) - im aware the transformer coil on the board is only 5v/0.5a and might not be powerful enough, but, i know the transformer accompanies the control chip on the underside of the pcb - im unsure if i can just swap it out to a 5v/1a 🤷🏼‍♂️

Then, i plan to use the buzzer output (as its 5v?) as the pwm control pin, and configure tasmota so i can set the buzzer output

ideally, the 5v power for the pwm controller will come from a GPIO so i can turn it on and off at the same time as the light relay, so the pwm controller isnt powered up unless the light is

my goal is to set a % via mqtt that will dim the light connected to the relay, via the pwm module sat in the middle

If using the buzzer pin (to get a full 5v) isnt necessary thats ok,

please fire any advice/tips, and i’ll go back and re-read all 200 comments on the “ifan03” issue again lol - theres a ton of handy info about the board

**Comment by djpimp2010** [comment]
Pic of the offending controller
![865E778A-577E-4EFC-9014-576B857D1FC0](https://user-images.githubusercontent.com/2423623/127642663-ee69df9b-078e-47de-bf40-6aa3aaffba9a.jpeg)

**Comment by Jason2866** [comment, marked as answer]
First of all, have you tested if the dimming board does dim the device like it should.
Connect it to a NodeMCU and do the setup you have in mind / want.
If this is working next step is to try to add to a modded ifan

**Reply by glennmurley15** [reply]
Yes, test on raspberry pi

Was controlling 3.3v power in via a GPIO HIGH / LOW
PWM connected to hardware GPIO pin,

To use the dimmer module, you need to power it up, and wait roughly two seconds, before sending first PWM power command (it also doesnt like power off before PWM off - it seems the PWM circuit is seperate from the power circuit, makes sense, the opto is seperating the two, and the PWM signal powers the input side of the opto, and the 3.3v in powers the opto output side of the circuit where the load is)

Initially controlled a filament bulb, with brightness levels, all seemed to work well, once I set the PWM frequency to 100Hz

The module doesnt switch off the bulb if the PWM signal is turned off (set to 0) and doesnt like the first PWM value to be higher than 90%

But, I've successfully controlled a filament bulb's brightness, without flickering, 
```
#new_val min=0.01, max=1
pi = pigpio.pi()
pi.hardware_PWM(12, 100, 1e6*new_val)
```

I was curious so I chucked a cheap LED bulb in, and it worked brilliantly, setting PWM to 1% turned it off, and anything above 15% was enough to make it power on, with no flickering,.... had to stop at 90% because any higher and it caused random flickers (not related to PWM on/off timing) - again, it could be the module itself,..

But that's brilliant news, because the cheap bulb I tried, is the ones I use around the house anyway, and the brightness levels between 20/90% were more than adequate for the room this will be installed in

So yes, to do this correctly, I need to permenantly wire the 3.3v power/ground to the dimmer module, to the sonoff ifan circuit, it must always be on, so that makes this easier, as I dont need to switch the power for the module

I am guessing to be able to send "full power" command through the module, I will need to definitely use a 5v signal for the PWM from the sonoff

I noticed when I had the multimeter on the buzzer circuit, there is always power in, and its the ground that connects to complete the buzzer circuit and make it sound off

This could all be irrelevant if there is another location on the board thats capable of being a 5v PWM output

**Reply by djpimp2010** [reply]
Well, jumped in headfirst after seeing the template in the ifan03 issue where @arendst added support for the device, where somebody wanted to use TX/RX for inputs

added the template, set GPIO3 to PWM/1 and a slider appeared, and it JUST WORKS!!!

wow, thanks guys!!!
![67166E6F-E3C6-4AB9-A4CC-01CB8C97E7D3](https://user-images.githubusercontent.com/2423623/127753731-d7dd9d97-3f98-40d3-aa3e-66784992fbef.jpeg)

**Comment by djpimp2010** [comment]
One would call this a resounding success…..

(If you can ignore the zipties holding it together, as I didnt put smaller gauge cable for the light power/common, and there isnt much room between the top of the capacitors and the case to run chunky cables up to the big open area that houses the dimmer module)

![3B1DE04A-C2C0-4570-9BE4-853987742EFB](https://user-images.githubusercontent.com/2423623/127757446-72a7c16a-a613-4b64-a8d8-63b6062d8d81.jpeg)

![255D8436-0C0D-4033-860A-75A9ADC95B57](https://user-images.githubusercontent.com/2423623/127757454-76040082-61fa-48d0-acfb-662c918ad64c.jpeg)

![142C66A6-9D5B-4FA4-B7BB-29E437F1298A](https://user-images.githubusercontent.com/2423623/127757459-9cb9b8d7-daa4-4a14-9f90-a332afe0f59f.jpeg)

![94A64883-363E-4D5D-B517-20C60C43F162](https://user-images.githubusercontent.com/2423623/127757467-631ab871-ebba-427a-8a2a-8b47026da5fe.jpeg)

![FC0DC9AE-F2B7-4610-8BE2-5761F2783B34](https://user-images.githubusercontent.com/2423623/127757471-e4078dda-59c2-4b7c-a397-6f8f08b2d973.jpeg)

![8270FACC-3CE6-46ED-BBC5-AC047F31147D](https://user-images.githubusercontent.com/2423623/127757475-a9f193ee-8aa3-4f6b-a070-7317161e4c30.jpeg)

**Reply by djpimp2010** [reply]
Yep, thats a win for me 🤷🏼‍♂️😎😎😎

![00AFE72B-8294-484A-8E1D-CF4E44EF74F5](https://user-images.githubusercontent.com/2423623/127759261-011ff347-1155-4fae-af33-1ae30f4a595f.png)
